tests/components/rest/test_binary_sensor.py

```diff
@@ -500,3 +500,27 @@ async def test_entity_config(hass: HomeAssistant) -> None:
         "friendly_name": "REST Binary Sensor",
         "icon": "mdi:one_two_three",
     }
+
+
+@respx.mock
+async def test_availability_in_config(hass: HomeAssistant) -> None:
+    """Test entity configuration."""
+
+    config = {
+        BINARY_SENSOR_DOMAIN: {
+            # REST configuration
+            "platform": DOMAIN,
+            "method": "GET",
+            "resource": "http://localhost",
+            # Entity configuration
+            "availability": "{{value==1}}",
+            "name": "{{'REST' + ' ' + 'Binary Sensor'}}",
+        },
+    }
+
+    respx.get("http://localhost") % HTTPStatus.OK
+    assert await async_setup_component(hass, BINARY_SENSOR_DOMAIN, config)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.rest_binary_sensor")
+    assert state.state == STATE_UNAVAILABLE
```

Comment:
Is this a new option or it was just not tested before?

Reply:
New options. Will open doc PR